File: ThefieldofdreamsClient/src/app/models/qr-code.ts
export type QRCodeType = 'Tourism' | 'Sponsor' | 'Mission' | 'Reward' | 'Checkpoint' | 'HiddenGem';

export interface QRCode {
  id: string;
  code: string;
  title: string;
  description?: string;
  type: QRCodeType;
  sponsorId?: string;
  locationId?: string;
  pointsReward: number;
  maxScans: number;
  currentScanCount: number;
  expiryDate?: string;
  isLimitedTime: boolean;
}

export interface CreateQRCodeRequest {
  code: string;
  title: string;
  description?: string;
  type: QRCodeType;
  sponsorId?: string;
  locationId?: string;
  pointsReward: number;
  maxScans: number;
  expiryDate?: string;
  isLimitedTime: boolean;
}

export interface UpdateQRCodeRequest {
  title: string;
  description?: string;
  type: QRCodeType;
  sponsorId?: string;
  locationId?: string;
  pointsReward: number;
  maxScans: number;
  expiryDate?: string;
  isLimitedTime: boolean;
}

export interface QRScan {
  id: string;
  userId: string;
  qrCodeId: string;
  pointsAwarded: number;
  locationData?: string;
  deviceInfo?: string;
  scannedAt: string;
}

export interface CreateQRScanRequest {
  qrCodeId: string;
  locationData?: string;
  deviceInfo?: string;
}

File: Thefieldofdreams.Domain/Entities/QRCode.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Thefieldofdreams.Domain.Entities
{

    public class QRCode : BaseEntity
    {
        public required string Code { get; set; }
        public required string Title { get; set; }
        public string? Description { get; set; }
        public QRCodeType Type { get; set; }
        public Guid? SponsorId { get; set; }
        public Guid? LocationId { get; set; }
        public int PointsReward { get; set; } = 10;
        public int MaxScans { get; set; } = 0; // 0 = unlimited
        public int CurrentScanCount { get; set; } = 0;
        public DateTime? ExpiryDate { get; set; }
        public bool IsLimitedTime { get; set; } = false;

        public virtual Sponsor? Sponsor { get; set; }
        public virtual Location? Location { get; set; }
        public virtual ICollection<QRCampaign> QRCampaigns { get; set; }

        public virtual ICollection<QRScan> QRScans { get; set; } = new List<QRScan>();
    }

    public enum QRCodeType
    {
        Tourism,
        Sponsor,
        Mission,
        Reward,
        Checkpoint,
        HiddenGem
    }
}
